From 6474c5cf84e5c972ea26ebb0eed1eecaec40d083 Mon Sep 17 00:00:00 2001
From: Dmitry Shachnev <mitya57@gmail.com>
Date: Sun, 31 Jan 2021 20:00:15 +0300
Subject: [PATCH] Add support for building with SIP v6

In SIP v6, sipName argument was removed from generateCode function and
added to set_globals.

See this changeset:
https://riverbankcomputing.com/hg/sip/rev/a16b972e11a1
---
 buildtools/wxpysip.py | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/buildtools/wxpysip.py b/buildtools/wxpysip.py
index c58102b84..be66e1966 100644
--- a/buildtools/wxpysip.py
+++ b/buildtools/wxpysip.py
@@ -62,8 +62,12 @@ def sip_runner(
         abi_major, abi_minor = resolve_abi_version(abi_version).split('.')
 
         # Set the globals.
-        set_globals(SIP_VERSION, SIP_VERSION_STR, int(abi_major), int(abi_minor),
-                UserException, include_dirs)
+        if (SIP_VERSION >> 16) >= 6:
+            set_globals(SIP_VERSION, SIP_VERSION_STR, int(abi_major), int(abi_minor),
+                    sip_module, UserException, include_dirs)
+        else:
+            set_globals(SIP_VERSION, SIP_VERSION_STR, int(abi_major), int(abi_minor),
+                    UserException, include_dirs)
 
         # Parse the input file.
         pt, _, _, _, tags, disabled_features = parse(specification,
@@ -72,9 +76,14 @@ def sip_runner(
 
         # Generate the bindings.
         if sources_dir is not None:
-            generated_files = generateCode(pt, sources_dir, source_suffix,
-                    exceptions, tracing, release_gil, parts, tags,
-                    disabled_features, docstrings, py_debug, sip_module)
+            if (SIP_VERSION >> 16) >= 6:
+                generated_files = generateCode(pt, sources_dir, source_suffix,
+                        exceptions, tracing, release_gil, parts, tags,
+                        disabled_features, docstrings, py_debug)
+            else:
+                generated_files = generateCode(pt, sources_dir, source_suffix,
+                        exceptions, tracing, release_gil, parts, tags,
+                        disabled_features, docstrings, py_debug, sip_module)
 
         if sbf_file is not None:
             generateBuildFile(sbf_file, generated_files)
